css-overflow-4/Overview.bs

```diff
@@ -2118,13 +2178,75 @@ showing in which case space is reserved for the <a>scrollbar gutter</a>.
+	represents a ''scroll marker control'' that forms a 'scroll marker group' for its nearest [=scroll container=]
+	in which exactly one control in the group can have its 'checked' state set to true.
+
+	The 'scroll marker group' that contains a ''&lt;button>'' element a also contains all the other ''&lt;button>'' elements b that fulfill the following conditions:
+
+	* The ''&lt;button>'' element b has a non-null ''HTMLButtonElement.scrollTargetElement'' value.
+	* a and b's ''HTMLButtonElement.scrollTargetElement'' have the same nearest [=scroll container=].
+
+	When a ''scroll marker control'' is activated:
+
+	1. Let <var>element</var> be the ''HTMLButtonElement.scrollTargetElement'' of the control.
+	1. Let <var>block</var> be "<code>start</code>".
+	1. Let <var>inline</var> be "<code>start</code>".
+	1. <a lt='scroll a target into view'>Scroll the element into view</a> with <var>behavior</var>, <var>block</var>, and <var>inline</var>.
+	1. If activated by invocation, move focus to <var>element</var>.
+		If <var>element</var> is not focusable this will result in there being no active element,
+		but the next focus change will proceed from this <var>element</var> as if it were focused.
+
+<h4 id="scroll-container-scroll">Scroll tracking</h4>
+
+	A scrolling operation might animate towards a particular position
+	(e.g. scrollbar arrow clicks, arrow key presses, "behavior: smooth" programmatic scrolls)
+	or might directly track a user’s input
+	(e.g. touch scrolling, scrollbar dragging).
+	In either case, the user agent chooses an 'eventual scroll position' to which the scroller
+	will reach.
+
+	Whenever a [=scroll container=] is scrolled, or layout changes the scroll position, the user agent must run these steps:
+
+	1. Let <var>position</var> be the 'eventual scroll position' for the scrolling operation.
+	1. Let <var>markers</var> be all of the ''scroll marker control'' elements which are a part of the 'scroll marker group' for the [=scroll container=].
+	1. Let <var>targets</var> be the set of ''HTMLButtonElement.scrollTargetElement'' for those controls sorted in [=tree order=].
+	1. Let <var>selected</var> be null.
+	1. For each item <var>target</var> in <var>targets</var>:
+		1. Set <var>selected</var> to <var>target</var>.
+		1. Let <var>targetPosition</var> be the position that would be scrolled to if we scroll <var>target</var> into view.
+		1. If <var>targetPosition</var>'s scroll block offset is less than or equal to <var>position</var>'s scroll block offset, break.
+		1. If <var>targetPosition</var>'s scroll inline offset is less than or equal to <var>position</var>'s scroll inline offset, break.
+	1. Let <var>marker</var> be the first control in <var>markers</var> whose ''HTMLButtonElement.scrollTargetElement'' is <var>selected</var>.
+	1. Set the 'checked' state of <var>marker</var> to true.
+
+<h4 id="scroll-target-focus">Focus behavior</h4>
+
+	A ''scroll marker control'' is only focusable if it is 'checked'.
```

Comment:
Is there any way to detect which button or pseudo-element is checked besides running a `querySelector()`? (I don't know if there needs to be, just curious.)

Reply:
We could always add this. It may make sense to find the currently active marker from the scroll container since that is what forms the exclusive group.

Reply:
Yeah, if we provide the API at all I think it would be on the scroll container.